### PR TITLE
Move env files to root and make detection clever

### PR DESCRIPTION
**Pull request by aronerben**
https://github.com/oxidizing/sihl/issues/291
https://github.com/oxidizing/sihl/issues/288

### CODE REVIEW

File: sihl/core/configuration.ml [hunk 1]
```diff
@@ -106,29 +104,52 @@ let is_production () =
...
+        let* line = Lwt_io.read_line_opt file in
+        match line with
+        | Some line -> read_to_end file (line :: ls)
+        | None -> Lwt.return ls
+      in
+      let* envs = read_to_end file [] in
+      envs |> envs_to_kv |> Lwt.return)
+    else (
+      Logs.debug (fun m -> m "Env file not found: %s. Continuing without it." filename);
+      Lwt.return [])
```
Comment (jerben) [inline sihl/core/configuration.ml:146]: @aronerben Returning `[]` implies that there is a env file that is empty. I'd return `None` probably, so the consumer function can decide to use a default `[]` but also can differentiate between non existing and empty.

File: test/unit/core_configuration.ml [hunk 2]
```diff
@@ -101,14 +101,20 @@ let read_env_file_non_existing _ () =
...
-  let filename = Sihl.Core.Configuration.project_root_path ^ "/.env.testing" in
-  Lwt_switch.add_hook (Some switch) (fun () -> Lwt_unix.unlink filename);
+  let* cur = Lwt_unix.getcwd () in
+  let marker = cur ^ "/.git" in
+  let filename = cur ^ "/.env.test" in
+  Lwt_switch.add_hook (Some switch) (fun () ->
+      let* () = Lwt_unix.unlink filename in
+      Lwt_unix.rmdir marker);
   let keys = [ "NAME"; "OCCUPATION"; "AGE" ] in
   let values = [ "church"; "mathematician"; "92" ] in
```
Comment (jerben) [inline test/unit/core_configuration.ml:116]: Props for proper age.

Comment (aronerben) [reply test/unit/core_configuration.ml:116]: :sunglasses:

File: sihl/core/configuration.ml (hunk 1 above)
Comment (aronerben) [reply sihl/core/configuration.ml]: Fixed!

**Pull request merged**